sklearn/preprocessing/tests/test_label.py

```diff
@@ -697,3 +706,36 @@ def test_label_encoders_do_not_have_set_output(encoder):
     y_encoded_with_kwarg = encoder.fit_transform(y=["a", "b", "c"])
     y_encoded_positional = encoder.fit_transform(["a", "b", "c"])
     assert_array_equal(y_encoded_with_kwarg, y_encoded_positional)
+
+
+@pytest.mark.parametrize(
+    "array_namespace, device, dtype", yield_namespace_device_dtype_combinations()
+)
+@pytest.mark.parametrize(
+    "y",
+    [
+        np.array([2, 1, 3, 1, 3]),
+        np.array([1, 1, 4, 5, -1, 0]),
+        np.array([3, 5, 9, 5, 9, 3]),
+    ],
+)
+def test_label_encoder_array_api_compliance(y, array_namespace, device, dtype):
```

Comment:
More precisely we cannot reuse our existing `sklearn.utils.estimator_checks.check_array_api_input` because it's written for classifiers or regressors and `LabelEncoder` is neither. Furthermore label encoder is kind of special because it requires integer inputs only.

So I think it's ok to not attempt to use a generic estimator check on this class and instead use a dedicated test as done in this PR.

Reply:
Olivier managed to put it into much better words than I did!